### ISSUE TITLE
Core/Loot: Items and references are not calculated correctly inside the same group

### ISSUE THREAD

**Issue by Killyana** [TrinityCore/TrinityCore#23908]
**Description:** If we put an item and a reference inside the same loot group the calculation will not be the sum of both entries.
Example:
Start by executing:
``` sql
UPDATE `reference_loot_template` SET GroupId=1, `MinCount`=2, `MaxCount`=2 WHERE `entry` = 12009 AND `Item` = 36908;
```
-- gameobject_loot_template
Entry, Item, Reference, Chance, QuestRequired, LootMode, GroupId, MinCount, MaxCount
24228, 1, 12009, 5, 0, 1, 1, 1, 1
24228, 36908, 0, 95, 0, 1, 1, 1, 1
-- reference_loot_template
Entry, Item, Reference, Chance, QuestRequired, LootMode, GroupId, MinCount, MaxCount
12009, 36908, 0, 100, 0, 1, 1, 2, 2

What must happen  in this case is the gob=190176 must always contain 95% 1 Lotus  or 5% 2 lotus.

What happen instead, the gob could contain 0 Lotus.

To have better representation of the issue we will change the drop rate to 50% 50%
``` sql
UPDATE `gameobject_loot_template` SET `Chance` = 50 WHERE `Entry` = 24228 AND `Item` IN (36908,1);
```
Result:
Entry, Item, Reference, Chance, QuestRequired, LootMode, GroupId, MinCount, MaxCount
24228, 1, 12009, 50, 0, 1, 1, 1, 1
24228, 36908, 0, 50, 0, 1, 1, 1, 1

And the result in game in this case:
1/3 1 lotus
1/3 2 lotus
1/3 0 lotus

So the issue is when you put and item and a reference inside a group the drop rate will not be the sum of the 2 but every rate is calculated separately, this could result having no item from this group (having 2 items from the same group is not possible)

**Steps to reproduce the problem:**

1. Execute the 2 sql above 
2. connect a char with 450 herbalism
3. .gob add temp 190176 and look at the number of lotus inside
4. do that many times and notice 1/3 of gob doesn't contain any lotus

**Branch(es):** both

**TC rev. hash/commit:** rev. c79571a09678

**Comment by Killyana** [comment]
If I set the reference to 50% and the item to 0 (it means 100 - (all drop rate of other items+refs from the group) the drop will be correct.
``` sql
UPDATE `gameobject_loot_template` SET `Chance` = 50 WHERE `Entry` = 24228 AND `Item` IN (1);
UPDATE `gameobject_loot_template` SET `Chance` = 0 WHERE `Entry` = 24228 AND `Item` IN (36908);

**Comment by Shauren** [comment]
This is working as intended, setting `GroupId` on a reference entry already has a special meaning - if the referenced loot has multiple groups in it, only process the one specified on parent
For example, if your reference_loot_template 12009 had one more item in it with `GroupId`=2, it would never drop

**Comment by Killyana** [comment]
Yes, but why the `Chance` must be 0 for the item 36908 to have the correct result, 50 also must give the same result as 0 in this case

**Comment by Shauren** [comment]
Loot groups are split into 2 containers, explicitly chanced (case 50) and equal chanced (case 0). When drops are determined, first all explicitly chanced items are rolled for "the one", if nothing is found, it falls  back to taking from equal chanced container

**Comment by meji46** [comment]
> This is working as intended, setting `GroupId` on a reference entry already has a special meaning - if the referenced loot has multiple groups in it, only process the one specified on parent
> For example, if your reference_loot_template 12009 had one more item in it with `GroupId`=2, it would never drop

Is there any situation in which this behavior is useful? (Apart from reducing number of reference entries)

**Comment by offl** [comment]
One more case https://wow.gamepedia.com/Void_Crystal#Source

> Epic equipment, iLevel 95-100: 50% x1, 50% x2

This is already implemented simply by mincount 1 and maxcount 2
SELECT * FROM disenchant_loot_template WHERE entry IN (66)

> Epic equipment, iLevel 105+: 33% x1, 67% x2

This does not work at all / badly implemented
SELECT * FROM disenchant_loot_template WHERE entry IN (67)
